crates/core/tedge_api/src/event.rs

```diff
@@ -48,6 +48,9 @@ pub mod error {
             error: String,
             payload: String,
         },
+
+        #[error("Failed to parse the payload to string that was received on topic: {topic}, due to error: {error}.")]
+        FailedToParsePayloadToString { topic: String, error: String },
```

Comment:
This error message is unclear for a non rust developer.

```suggestion
        #[error("Failed to parse as an UTF-8 string the payload received on topic: {topic}, due to error: {error}.")]
        FailedToParsePayloadToString { topic: String, error: String },
```